docs/table.md

```diff
@@ -0,0 +1,194 @@
+  | `visibleRows: T[]`                                                                                     | The currently visible rows.
+  | `allVisibleRowsSelected: boolean`                                                                      | Indicates if all visible rows are selected.
+  | `loading: boolean`                                                                                     | Indicates if the table is currently loading data.
+  | `error: any`                                                                                           | The error that occured in the last observable returned by loadData.
+  | `locale: string`                                                                                       | The locale for the table texts.
+  | `dataLength: number`                                                                                   | The length of the total number of items that are being paginated.
+  | `sortColumn: string`                                                                                   | The name of the column, after which the rows should be sorted.
+  | `sortDirection: 'asc' | 'desc'`                                                                        | The sort direction.
+  | `pageIndex: number`                                                                                    | The zero-based page index of the displayed list of items.
+  | `pageSize: number`                                                                                     | Number of items to display on a page.
+  | `filter: string`                                                                                       | Filter term that should be used to filter out objects from the data array. To override how data objects match to this filter string, provide a custom function for filterPredicate.
+  | `filterPredicate: () => boolean`                                                                       | Checks if a data object matches the data source's filter string. By default, each data object is converted to a string of its properties and returns true if the filter has at least one occurrence in that string. By default, the filter string has its whitespace trimmed and the match is case-insensitive. May be overridden for a custom implementation of filter matching.
+  | `sortingDataAccessor: (data: T[], sort: { sortColumn: string; sortDirection: 'asc' | 'desc' }) => T[]` | Gets a sorted copy of the data array based on the state of the MatSort. Called after changes are made to the filtered data or when sort changes are emitted from MatSort. By default, the function retrieves the active sort and its direction and compares data by retrieving data using the sortingDataAccessor. May be overridden for a custom implementation of data ordering.

```

Comment:
this is the function signature and description for sortData

But the doc for sortingDataAccessor is missing which has the signature: (data: T, columnName: string) => any
Data accessor function that is used for accessing data properties for sorting through the default sortData function. This default function assumes that the sort header IDs (which defaults to the column name) matches the data's properties (e.g. column Xyz represents data['Xyz']). May be set to a custom function for different behavior.